packages/actor-iterator-transform-record-intermediate-results/lib/ActorIteratorTransformRecordIntermediateResults.ts

```diff
@@ -0,0 +1,62 @@
+import type {
+  IActionIteratorTransformBindings,
+  IActionIteratorTransformQuads,
+  IActorIteratorTransformArgs,
+  ITransformIteratorOutput }
+  from '@comunica/bus-iterator-transform';
+import { ActorIteratorTransform } from '@comunica/bus-iterator-transform';
+import { KeysStatistics } from '@comunica/context-entries';
+import type { StatisticIntermediateResults } from '@comunica/statistic-intermediate-results';
+import type { MetadataBindings, MetadataQuads } from '@comunica/types';
+import type * as RDF from '@rdfjs/types';
+import type { AsyncIterator } from 'asynciterator';
+
+/**
+ * A comunica Record Intermediate Results Iterator Transform Actor.
+ * This actor updates the intermediate result statistic when an intermediate result is produced.
+ */
+export class ActorIteratorTransformRecordIntermediateResults extends ActorIteratorTransform {
+  public constructor(args: IActorIteratorTransformArgs) {
+    super(args);
+  }
+
+  public async transformIteratorBindings(action: IActionIteratorTransformBindings):
+  Promise<ITransformIteratorOutput<AsyncIterator<RDF.Bindings>, MetadataBindings>> {
+    const statisticIntermediateResults: StatisticIntermediateResults = action.context
+      .getSafe(KeysStatistics.intermediateResults);
+    const output = action.stream.map((data) => {
+      statisticIntermediateResults.updateStatistic(
+        {
+          type: action.type,
+          data,
+          metadata: {
+            operation: action.operation,
+            metadata: action.metadata,
+          },
+        },
+      );
+      return data;
+    });
+    return { stream: output, metadata: action.metadata };
+  }
```

Comment:
@rubensworks I'm not sure about this one. Should we leave the `metadata` call to the statistic or do it in the `transformIterator` call? I'm leaning towards calling it in the statistic, as statistics might not use the metadata so it would increase overhead for no reason.

Reply:
I agree with your opinion @RubenEschauzier, let's leave calling it to the statistic itself.